tests/shell_redirection_tests.cpp

```diff
@@ -39,152 +39,455 @@
+	                                                 .is_appending = true};
+
+	// verified in MS-DOS 6.22
+	test_redirection(line, expected_results);
+}
+
+TEST(DOS_Shell_GetRedirection, TwoInputs)
+{
+	constexpr auto line = "more < in1.txt < in2.txt";
+
+	const auto expected_results = RedirectionResults{.processed_line = "more ",
+	                                                 .in_file = "in2.txt"};
+
+	// verified in MS-DOS 6.22 (last input wins)
+	test_redirection(line, expected_results);
+}
+
+TEST(DOS_Shell_GetRedirection, TwoInputsNoWhitespace)
+{
+	constexpr auto line = "more<in1.txt<in2.txt";
+
+	const auto expected_results = RedirectionResults{.processed_line = "more",
+	                                                 .in_file = "in2.txt"};
+
+	// verified in MS-DOS 6.22 (last input wins)
+	test_redirection(line, expected_results);
+}
+
+TEST(DOS_Shell_GetRedirection, TwoOutputsNoTrailingWhitespace)
+{
+	constexpr auto line = "echo test>out1.txt>out2.txt";
+
+	const auto expected_results = RedirectionResults{.processed_line = "echo test ",
+	                                                 .out_file = "out2.txt"};
+
+	// verified in MS-DOS 6.22: last output wins, and replaced prior
+	// directive is replaced with 1 space if the replaced directives have
+	// not trailing whitespcea. If the replaced directive has trailing white
+	// space, then it's replaced without the space however the trailing
+	// space is left in. (see next test).
+	test_redirection(line, expected_results);
+}
+
+TEST(DOS_Shell_GetRedirection, TwoOutputsNoTrailingWhitespace2)
+{
+	constexpr auto line = "echo test>    out1.txt>     out2.txt";
+
+	const auto expected_results = RedirectionResults{.processed_line = "echo test ",
+	                                                 .out_file = "out2.txt"};
+
+	// verified in MS-DOS 6.22
+	test_redirection(line, expected_results);
+}
+
+TEST(DOS_Shell_GetRedirection, TwoOutputsFirstTrailingSpace)
+{
+	constexpr auto line = "echo test>out1.txt >out2.txt";
+
+	const auto expected_results = RedirectionResults{.processed_line = "echo test ",
+	                                                 .out_file = "out2.txt"};
+
+	// verified in MS-DOS 6.22
+	test_redirection(line, expected_results);
+}
+
+TEST(DOS_Shell_GetRedirection, TwoOutputsFirstTrailingTwoSpaces)
+{
+	constexpr auto line = "echo test>out1.txt  >out2.txt";
+
+	const auto expected_results = RedirectionResults{.processed_line = "echo test  ",
+	                                                 .out_file = "out2.txt"};
+
+	// verified in MS-DOS 6.22
+	test_redirection(line, expected_results);
+}
+
+TEST(DOS_Shell_GetRedirection, TwoOutputsFirstBothTrailing)
+{
+	constexpr auto line = "echo test>out1.txt >out2.txt ";
+
+	const auto expected_results = RedirectionResults{.processed_line = "echo test  ",
+	                                                 .out_file = "out2.txt"};
+
+	// verified in MS-DOS 6.22
+	test_redirection(line, expected_results);
+}
```

Comment:
That's some very thorough test coverage and nice use of the designated initialisers 🚀